numojo/core/ndarray.mojo

```diff
@@ -859,33 +921,66 @@ struct NDArray[dtype: DType = DType.float32](Stringable):
     fn num_elements(self) -> Int:
         return self.info.size
 
+    # TODO: move this initialization to the Fields and constructor
     fn shape(self) -> NDArrayShape:
         var shapeNDArray: NDArrayShape = NDArrayShape(self.info.shape)
         return shapeNDArray
 
     fn load[width: Int](self, idx: Int) -> SIMD[dtype, width]:
+        """
+        Loads a SIMD element of size `width` at the given index `idx`.
+        """
         return self._arr.load[width=width](idx)
 
-    fn load[width:Int = 1](self, *indices:Int) -> SIMD[dtype, width]:
+    # TODO: we should add checks to make sure user don't load out of bound indices, but that will overhead, figure out later
```

Comment:
I think if we do that we should make a load_safe in addition to load, to avoid having to error handle everywhere we call load. This follows the philosophy that safety should be easy but not mandatory.

Reply:
ah! That would be a work around, I will include that in cleanup. What do you say for getters and setters similarly? I think we can keep getters and setters with all the checks and use load, store for fast and SIMD operations.

Reply:
I think load should behave like the unsafe pointer operation, and load_safe should be bounds checked and eventually be a safe pointer method.